File: src/Pos.Application/Dtos/OrderResponse.cs
namespace Pos.Application.Dtos;

public class OrderResponse
{
    public int Id { get; set; }

    public int UserId { get; set; }

    public DateTime PlacedAt { get; set; }

    public string Status { get; set; } = string.Empty;

    public string PaymentMethod { get; set; } = string.Empty;

    public decimal Total { get; set; }

    public List<OrderItemResponse> Items { get; set; } = new List<OrderItemResponse>();
}

File: src/web/lib/types/OrderResponse.ts
import type { OrderItemResponse } from "@/lib/types/OrderItemResponse";

// An order as the API returns it.
export type OrderResponse = {
  id: number;
  userId: number;
  placedAt: string;
  status: string;
  paymentMethod: string;
  total: number;
  items: OrderItemResponse[];
};
